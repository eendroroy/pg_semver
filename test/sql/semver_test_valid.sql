LOAD 'pg_semver';

CREATE TABLE versions(version SEMVER);

INSERT INTO versions VALUES ('0');
INSERT INTO versions VALUES ('0.0');
INSERT INTO versions VALUES ('0.1');
INSERT INTO versions VALUES ('0.0.0');
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.0.1-alpha.1');
INSERT INTO versions VALUES ('0.0.1-beta.1');
INSERT INTO versions VALUES ('0.0.1-rc.1');



INSERT INTO versions VALUES ('!0');
INSERT INTO versions VALUES ('@0.0');
INSERT INTO versions VALUES ('0.#1');
INSERT INTO versions VALUES ('0.0.%0');
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.&0.1-alpha.0');

SELECT * FROM versions;

DROP TABLE IF EXISTS versions;
