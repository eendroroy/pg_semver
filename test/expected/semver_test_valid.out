LOAD 'pg_semver';
CREATE TABLE versions(version SEMVER);
INSERT INTO versions VALUES ('0');
INSERT INTO versions VALUES ('0.0');
INSERT INTO versions VALUES ('0.1');
INSERT INTO versions VALUES ('0.0.0');
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.0.1-alpha.1');
INSERT INTO versions VALUES ('0.0.1-beta.1');
INSERT INTO versions VALUES ('0.0.1-rc.1');
INSERT INTO versions VALUES ('!0');
ERROR:  Invalid Semver '!0'.
LINE 1: INSERT INTO versions VALUES ('!0');
                                     ^
DETAIL:  Semver '!0' contains invalid character.
HINT:  Did you mean '0'?
INSERT INTO versions VALUES ('@0.0');
ERROR:  Invalid Semver '@0.0'.
LINE 1: INSERT INTO versions VALUES ('@0.0');
                                     ^
DETAIL:  Semver '@0.0' contains invalid character.
HINT:  Did you mean '0.0'?
INSERT INTO versions VALUES ('0.#1');
ERROR:  Invalid Semver '0.#1'.
LINE 1: INSERT INTO versions VALUES ('0.#1');
                                     ^
DETAIL:  Semver '0.#1' contains invalid character.
HINT:  Did you mean '0.1'?
INSERT INTO versions VALUES ('0.0.%0');
ERROR:  Invalid Semver '0.0.%0'.
LINE 1: INSERT INTO versions VALUES ('0.0.%0');
                                     ^
DETAIL:  Semver '0.0.%0' contains invalid character.
HINT:  Did you mean '0.0.0'?
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.&0.1-alpha.0');
ERROR:  Invalid Semver '0.&0.1-alpha.0'.
LINE 1: INSERT INTO versions VALUES ('0.&0.1-alpha.0');
                                     ^
DETAIL:  Semver '0.&0.1-alpha.0' contains invalid character.
HINT:  Did you mean '0.0.1-alpha.0'?
SELECT * FROM versions;
    version    
---------------
 0
 0.0
 0.1
 0.0.0
 0.0.1
 0.0.1-alpha.1
 0.0.1-beta.1
 0.0.1-rc.1
 0.0.1
(9 rows)

DROP TABLE IF EXISTS versions;
